File: docs/examples/go/add_ping_service.go
// Command add_ping_service registers a new ping check for the project.
package main

import (
    "bytes"
    "encoding/json"
    "fmt"
    "io"
    "net/http"
)

const (
    pingToken = "<token>"
    pingProjectUUID = "<project-uuid>"
)

func main() {
    payload := map[string]any{
        "name":         "Billing health-check",
        "url":          "https://billing.example.com/health",
        "interval":     60,
        "telegramTags": []string{"PING_DOWN"},
    }

    body, _ := json.Marshal(payload)
    req, _ := http.NewRequest(http.MethodPost, baseURL+"/projects/"+pingProjectUUID+"/ping-services", bytes.NewReader(body))
    req.Header.Set("Content-Type", "application/json")
    req.Header.Set("Authorization", "Bearer "+pingToken)

    resp, err := http.DefaultClient.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusCreated {
        data, _ := io.ReadAll(resp.Body)
        panic(fmt.Sprintf("unexpected status %d: %s", resp.StatusCode, data))
    }

    var service map[string]any
    if err := json.NewDecoder(resp.Body).Decode(&service); err != nil {
        panic(err)
    }

    fmt.Println("Created ping service:", service)
}

File: docs/examples/python/add_ping_service.py
"""Register a ping service for a project."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"
PROJECT_UUID = "<project-uuid>"


def add_ping_service() -> None:
    payload = {
        "name": "Billing health-check",
        "url": "https://billing.example.com/health",
        "interval": 60,
        "telegramTags": ["PING_DOWN"],
    }
    response = requests.post(
        f"{BASE_URL}/projects/{PROJECT_UUID}/ping-services",
        json=payload,
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    print("Created ping service:", response.json())


if __name__ == "__main__":
    add_ping_service()
